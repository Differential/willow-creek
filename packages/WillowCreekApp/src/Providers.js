import { nest } from 'recompose';
import { Providers } from '@apollosproject/ui-kit';
import ClientProvider from './client';

export default nest(ClientProvider, Providers);
