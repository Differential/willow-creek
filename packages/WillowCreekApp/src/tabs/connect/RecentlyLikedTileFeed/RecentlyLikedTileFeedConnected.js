import React from 'react';
import { Query } from 'react-apollo';
import PropTypes from 'prop-types';

import getLikedContent from '../getLikedContent';
import RecentlyLikedTileFeed from './RecentlyLikedTileFeed';

const RecentlyLikedTileFeedConnected = ({ navigation }) => (
  <Query query={getLikedContent} fetchPolicy="cache-and-network">
    {({ loading, data: { getAllLikedContent = [] } = {} }) => {
      if (!getAllLikedContent.length) return null;
      return (
        <RecentlyLikedTileFeed
          id={'liked'}
          name={'Recently Liked'}
          content={getAllLikedContent}
          isLoading={loading}
          navigation={navigation}
          loadingStateObject={{
            title: 'Recently Liked',
            isLoading: true,
          }}
        />
      );
    }}
  </Query>
);

RecentlyLikedTileFeedConnected.propTypes = {
  navigation: PropTypes.shape({
    getParam: PropTypes.func,
    navigate: PropTypes.func,
  }),
};

export default RecentlyLikedTileFeedConnected;
