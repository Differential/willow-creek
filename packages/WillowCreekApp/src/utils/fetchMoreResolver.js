import get from 'lodash/get';
import set from 'lodash/fp/set';

export default function fetchMoreResolver({
  collectionName,
  variables,
  fetchMore,
  data,
} = {}) {
  return () => {
    const pageInfoPath = `${collectionName}.pageInfo`;
    const edgePath = `${collectionName}.edges`;

    const after = get(data, `${pageInfoPath}.endCursor`);
    if (!after) return;

    fetchMore({
      variables: { ...variables, after },
      updateQuery: (previousResult, { fetchMoreResult }) => {
        // combine previous data and fetchMore data
        const newDataWithPageInfo = set(
          pageInfoPath,
          {
            ...get(previousResult, pageInfoPath, {}),
            ...get(fetchMoreResult, pageInfoPath, {}),
          },
          previousResult
        );
        const newDataWithAdditionalItems = set(
          edgePath,
          [
            ...get(previousResult, edgePath, []),
            ...get(fetchMoreResult, edgePath, []),
          ],
          newDataWithPageInfo
        );

        return newDataWithAdditionalItems;
      },
    });
  };
}
