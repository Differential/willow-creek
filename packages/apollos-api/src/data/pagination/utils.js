// eslint-disable-next-line import/prefer-default-export
export const withEdgePagination = async ({ edges }) => {
  const result = await edges;
  const startCursor = result.length ? result[0].cursor : null;
  const endCursor = result.length ? result[result.length - 1].cursor : null;
  return { startCursor, endCursor };
};
