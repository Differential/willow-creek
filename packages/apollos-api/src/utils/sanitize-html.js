import sanitizeHtml from 'sanitize-html';

const allowedTags = [
  'h1',
  'h2',
  'h3',
  'h4',
  'h5',
  'h6',
  'blockquote',
  'p',
  'a',
  'ul',
  'ol',
  'li',
  'b',
  'i',
  'strong',
  'em',
  'br',
  'caption',
  'img',
];

const allowedAttributes = {
  a: ['href'],
  img: ['src'],
};

// A very picky HTML sanitizer
export default function(dirty) {
  return sanitizeHtml(dirty, {
    allowedTags,
    allowedAttributes,
  });
}
