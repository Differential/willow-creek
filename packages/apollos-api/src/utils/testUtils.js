import { KeyValueCache } from 'apollo-server-caching';
import {
  context as getContext,
  dataSources as getDataSources,
} from '../data/index';

export function getTestContext(req) {
  const context = getContext(req);
  const dataSources = getDataSources();
  // Apollo Server does this internally.
  Object.values(dataSources).forEach((dataSource) => {
    if (dataSource.initialize) {
      dataSource.initialize({ context, cache: KeyValueCache });
    }
  });
  context.dataSources = dataSources;
  return context;
}

export const buildGetMock = (response, dataSource) => {
  const get = jest.fn();
  if (Array.isArray(response) && Array.isArray(response[0])) {
    response.forEach((responseVal) => {
      get.mockReturnValueOnce(
        new Promise((resolve) => resolve(dataSource.normalize(responseVal)))
      );
    });
  }
  get.mockReturnValue(
    new Promise((resolve) => resolve(dataSource.normalize(response)))
  );
  return get;
};
