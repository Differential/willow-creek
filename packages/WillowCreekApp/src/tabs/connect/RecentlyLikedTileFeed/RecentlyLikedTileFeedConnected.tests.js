import React from 'react';

import Providers from '../../../Providers';

import { renderWithApolloData } from '../../../utils/testUtils';

import GET_LIKED_CONTENT from '../getLikedContent';
import RecentlyLikedTileFeedConnected from './RecentlyLikedTileFeedConnected';

describe('RecentlyLikedTileFeedConnected', () => {
  it('renders a RecentlyLikedTileFeedConnected', async () => {
    const mock = {
      request: {
        query: GET_LIKED_CONTENT,
        variables: { first: 3 },
      },
      result: {
        data: {
          likedContent: {
            __typename: 'ContentItemsConnection',
            pageInfo: {
              __typename: 'PaginationInfo',
              endCursor: '123',
            },
            edges: [
              {
                __typename: 'ContentItemsConnectionEdge',
                node: {
                  __typename: 'UniversalContentItem',
                  id: 'UniversalContentItem:d57994350b9d213866b24dea3a97433d',
                  coverImage: null,
                  parentChannel: {
                    id: 'ContentChannel:4f68015ba18662a7409d1219a4ce013e',
                    name: 'Editorial',
                    iconName: 'text',
                    __typename: 'ContentChannel',
                  },
                  title: 'Mea Animal Aperiam Ornatus Eu',
                  hyphenatedTitle: 'Mea Animal Aperiam Ornatus Eu',
                  summary: 'Bla bla bla',
                  theme: null,
                  isLiked: false,
                  likedCount: 0,
                  videos: [],
                  audios: [],
                },
              },
              {
                __typename: 'ContentItemsConnectionEdge',
                node: {
                  __typename: 'UniversalContentItem',
                  id: 'UniversalContentItem:b36e55d803443431e96bb4b5068147ec',
                  coverImage: null,
                  parentChannel: {
                    id: 'ContentChannel:4f68015ba18662a7409d1219a4ce013e',
                    name: 'Editorial',
                    iconName: 'text',
                    __typename: 'ContentChannel',
                  },
                  title: 'Probo Senserit Id Mea, Ut Sed Malis Postea,',
                  hyphenatedTitle:
                    'Probo Senserit Id Mea, Ut Sed Malis Postea,',
                  summary: 'Bla bla bla',
                  theme: null,
                  isLiked: false,
                  likedCount: 0,
                  videos: [],
                  audios: [],
                },
              },
            ],
          },
        },
      },
    };
    const navigation = { navigate: jest.fn(), getParam: jest.fn() };
    const tree = await renderWithApolloData(
      <Providers mocks={[mock]}>
        <RecentlyLikedTileFeedConnected navigation={navigation} />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });

  it('renders nothing if no liked content', async () => {
    const mock = {
      request: {
        query: GET_LIKED_CONTENT,
      },
      result: {
        data: {
          likedContent: {
            __typename: 'ContentItemsConnection',
            pageInfo: {
              __typename: 'PaginationInfo',
              endCursor: null,
            },
            edges: [],
          },
        },
      },
    };
    const navigation = { navigate: jest.fn(), getParam: jest.fn() };
    const tree = await renderWithApolloData(
      <Providers mocks={[mock]}>
        <RecentlyLikedTileFeedConnected navigation={navigation} />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
});
