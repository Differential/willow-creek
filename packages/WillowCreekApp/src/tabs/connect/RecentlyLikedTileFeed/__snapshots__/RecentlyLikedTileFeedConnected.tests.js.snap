// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecentlyLikedTileFeedConnected renders a RecentlyLikedTileFeedConnected 1`] = `
<RCTSafeAreaView
  onLayout={[Function]}
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <View
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      horizontal={false}
      style={
        Object {
          "paddingHorizontal": 0,
          "paddingVertical": 16,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingHorizontal": 16,
            "paddingVertical": 16,
          }
        }
      >
        <View
          style={
            Object {
              "flexGrow": 1,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#303030",
                "fontFamily": "InterUI-Bold",
                "fontSize": 16,
                "lineHeight": 23.04,
              }
            }
          >
            Recently Liked
          </Text>
        </View>
        <View
          style={
            Object {
              "flexDirection": "row-reverse",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#303030",
                "fontFamily": "System",
                "fontSize": 14,
                "fontStyle": null,
                "fontWeight": null,
                "lineHeight": 20.16,
              }
            }
          >
            <Text
              onPress={[Function]}
              style={
                Object {
                  "color": "#17B582",
                }
              }
            >
              View All
            </Text>
          </Text>
        </View>
      </View>
      <RCTScrollView
        data={
          Array [
            Object {
              "__typename": "UniversalContentItem",
              "coverImage": Object {
                "sources": Array [
                  Object {
                    "uri": "https://apollosrock.newspring.cc/GetImage.ashx?guid=58c037fa-cc7f-4d72-9571-a0cc0558e346",
                  },
                ],
              },
              "id": "UniversalContentItem:296373ecb53580855cadffa0375ebe18",
              "title": "Guys Night!",
            },
            Object {
              "__typename": "UniversalContentItem",
              "coverImage": Object {
                "sources": Array [
                  Object {
                    "uri": "https://apollosrock.newspring.cc/GetImage.ashx?guid=f54b0db0-95f5-44ad-b8f2-8bcd1b23cfdb",
                  },
                ],
              },
              "id": "UniversalContentItem:39193b5ad28717ebfeab4d226664d8e7",
              "title": "This August at NewSpring",
            },
            Object {
              "__typename": "UniversalContentItem",
              "coverImage": Object {
                "sources": Array [
                  Object {
                    "uri": "https://apollosrock.newspring.cc/GetImage.ashx?guid=a65bc45d-f961-4b7e-a899-63eb1f9b8da9",
                  },
                ],
              },
              "id": "UniversalContentItem:ae8ec75906ba7437c49ad2534b5024db",
              "title": "A Place to Worship Free of Fear",
            },
          ]
        }
        decelerationRate="fast"
        disableVirtualization={false}
        getItem={[Function]}
        getItemCount={[Function]}
        horizontal={true}
        initialNumToRender={5}
        initialScrollIndex={0}
        keyExtractor={[Function]}
        loadingStateObject={
          Object {
            "coverImage": Array [],
            "id": "fake_id",
            "title": "",
          }
        }
        maxToRenderPerBatch={10}
        numColumns={1}
        onContentSizeChange={[Function]}
        onEndReachedThreshold={2}
        onLayout={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        refreshing={false}
        renderItem={[Function]}
        scrollEventThrottle={50}
        showsHorizontalScrollIndicator={false}
        snapToAlignment="start"
        snapToInterval={592}
        stickyHeaderIndices={Array []}
        style={
          Object {
            "height": 600,
            "paddingBottom": 8,
          }
        }
        tileHeight={600}
        updateCellsBatchingPeriod={50}
        viewabilityConfigCallbackPairs={Array []}
        windowSize={21}
      >
        <View>
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "flexDirection": "row",
                },
                null,
              ]
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                  "padding": 8,
                  "paddingLeft": 16,
                  "paddingRight": 0,
                }
              }
            >
              <View
                accessible={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "transform": Array [
                      Object {
                        "scale": 1,
                      },
                    ],
                  }
                }
              >
                <View
                  style={
                    Object {
                      "aspectRatio": 1,
                      "borderRadius": 16,
                      "height": "100%",
                      "overflow": "hidden",
                      "width": "100%",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "backgroundColor": "#A5A5A5",
                        "width": "100%",
                      }
                    }
                  >
                    <View
                      animate={null}
                      customAnimate={null}
                      onReady={false}
                      style={
                        Object {
                          "aspectRatio": 1,
                          "backgroundColor": "#A5A5A5",
                          "width": "100%",
                        }
                      }
                    >
                      <Image
                        onLoad={[Function]}
                        source={
                          Array [
                            Object {
                              "cache": "force-cache",
                              "uri": "https://apollosrock.newspring.cc/GetImage.ashx?guid=58c037fa-cc7f-4d72-9571-a0cc0558e346",
                            },
                          ]
                        }
                        style={
                          Object {
                            "aspectRatio": 1,
                            "backgroundColor": "#A5A5A5",
                            "opacity": 0,
                            "resizeMode": "cover",
                            "width": "100%",
                          }
                        }
                      />
                    </View>
                    <BVLinearGradient
                      colors={
                        Array [
                          0,
                          4278190080,
                        ]
                      }
                      endPoint={
                        Object {
                          "x": 0,
                          "y": 1,
                        }
                      }
                      locations={
                        Array [
                          0.3,
                          1,
                        ]
                      }
                      startPoint={
                        Object {
                          "x": 0,
                          "y": 0,
                        }
                      }
                      style={
                        Object {
                          "bottom": 0,
                          "left": 0,
                          "position": "absolute",
                          "right": 0,
                          "top": 0,
                        }
                      }
                    />
                  </View>
                  <Text
                    style={
                      Object {
                        "backgroundColor": "transparent",
                        "bottom": 16,
                        "color": "#F8F7F4",
                        "fontFamily": "InterUI-Bold",
                        "fontSize": 16,
                        "left": 16,
                        "lineHeight": 23.04,
                        "position": "absolute",
                        "right": 16,
                      }
                    }
                  >
                    Guys Night!
                  </Text>
                </View>
              </View>
            </View>
          </View>
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "flexDirection": "row",
                },
                null,
              ]
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                  "padding": 8,
                  "paddingLeft": 16,
                  "paddingRight": 0,
                }
              }
            >
              <View
                accessible={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "transform": Array [
                      Object {
                        "scale": 1,
                      },
                    ],
                  }
                }
              >
                <View
                  style={
                    Object {
                      "aspectRatio": 1,
                      "borderRadius": 16,
                      "height": "100%",
                      "overflow": "hidden",
                      "width": "100%",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "backgroundColor": "#A5A5A5",
                        "width": "100%",
                      }
                    }
                  >
                    <View
                      animate={null}
                      customAnimate={null}
                      onReady={false}
                      style={
                        Object {
                          "aspectRatio": 1,
                          "backgroundColor": "#A5A5A5",
                          "width": "100%",
                        }
                      }
                    >
                      <Image
                        onLoad={[Function]}
                        source={
                          Array [
                            Object {
                              "cache": "force-cache",
                              "uri": "https://apollosrock.newspring.cc/GetImage.ashx?guid=f54b0db0-95f5-44ad-b8f2-8bcd1b23cfdb",
                            },
                          ]
                        }
                        style={
                          Object {
                            "aspectRatio": 1,
                            "backgroundColor": "#A5A5A5",
                            "opacity": 0,
                            "resizeMode": "cover",
                            "width": "100%",
                          }
                        }
                      />
                    </View>
                    <BVLinearGradient
                      colors={
                        Array [
                          0,
                          4278190080,
                        ]
                      }
                      endPoint={
                        Object {
                          "x": 0,
                          "y": 1,
                        }
                      }
                      locations={
                        Array [
                          0.3,
                          1,
                        ]
                      }
                      startPoint={
                        Object {
                          "x": 0,
                          "y": 0,
                        }
                      }
                      style={
                        Object {
                          "bottom": 0,
                          "left": 0,
                          "position": "absolute",
                          "right": 0,
                          "top": 0,
                        }
                      }
                    />
                  </View>
                  <Text
                    style={
                      Object {
                        "backgroundColor": "transparent",
                        "bottom": 16,
                        "color": "#F8F7F4",
                        "fontFamily": "InterUI-Bold",
                        "fontSize": 16,
                        "left": 16,
                        "lineHeight": 23.04,
                        "position": "absolute",
                        "right": 16,
                      }
                    }
                  >
                    This August at NewSpring
                  </Text>
                </View>
              </View>
            </View>
          </View>
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "flexDirection": "row",
                },
                null,
              ]
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                  "padding": 8,
                  "paddingLeft": 16,
                  "paddingRight": 0,
                }
              }
            >
              <View
                accessible={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "transform": Array [
                      Object {
                        "scale": 1,
                      },
                    ],
                  }
                }
              >
                <View
                  style={
                    Object {
                      "aspectRatio": 1,
                      "borderRadius": 16,
                      "height": "100%",
                      "overflow": "hidden",
                      "width": "100%",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "backgroundColor": "#A5A5A5",
                        "width": "100%",
                      }
                    }
                  >
                    <View
                      animate={null}
                      customAnimate={null}
                      onReady={false}
                      style={
                        Object {
                          "aspectRatio": 1,
                          "backgroundColor": "#A5A5A5",
                          "width": "100%",
                        }
                      }
                    >
                      <Image
                        onLoad={[Function]}
                        source={
                          Array [
                            Object {
                              "cache": "force-cache",
                              "uri": "https://apollosrock.newspring.cc/GetImage.ashx?guid=a65bc45d-f961-4b7e-a899-63eb1f9b8da9",
                            },
                          ]
                        }
                        style={
                          Object {
                            "aspectRatio": 1,
                            "backgroundColor": "#A5A5A5",
                            "opacity": 0,
                            "resizeMode": "cover",
                            "width": "100%",
                          }
                        }
                      />
                    </View>
                    <BVLinearGradient
                      colors={
                        Array [
                          0,
                          4278190080,
                        ]
                      }
                      endPoint={
                        Object {
                          "x": 0,
                          "y": 1,
                        }
                      }
                      locations={
                        Array [
                          0.3,
                          1,
                        ]
                      }
                      startPoint={
                        Object {
                          "x": 0,
                          "y": 0,
                        }
                      }
                      style={
                        Object {
                          "bottom": 0,
                          "left": 0,
                          "position": "absolute",
                          "right": 0,
                          "top": 0,
                        }
                      }
                    />
                  </View>
                  <Text
                    style={
                      Object {
                        "backgroundColor": "transparent",
                        "bottom": 16,
                        "color": "#F8F7F4",
                        "fontFamily": "InterUI-Bold",
                        "fontSize": 16,
                        "left": 16,
                        "lineHeight": 23.04,
                        "position": "absolute",
                        "right": 16,
                      }
                    }
                  >
                    A Place to Worship Free of Fear
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`RecentlyLikedTileFeedConnected renders nothing if no liked content 1`] = `
<RCTSafeAreaView
  onLayout={[Function]}
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <View
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  />
</RCTSafeAreaView>
`;
